app.py

```diff
@@ -295,7 +296,19 @@ def valid_request(request: Dict[str, str]) -> str:
     return(auth_status)
 
 
-def get_container(userid: str, request: flask.Request, narrative: str) -> flask.Response:
+def clean_userid( userid: str) -> str:
+    """
+    Takes a normal KBase userid and converts it into a userid that is okay to embed in a rancher servicename
+    """
+    hash = hashlib.sha1(userid.encode()).hexdigest()
+    hash = hash[:6]
+    cleaned = re.sub('[\._-]+', '-', userid)
+    max_len = 63 - len(cfg['container_name']) - len(hash)
+    cleaned = cleaned[:max_len]+hash
```

Comment:
Should we insert a dash between the munged username and the hash?  Would make the service name more readable but we'd lose a character.

Reply:
Good catch on the narrative shutdown. I've added in the call to clean_userid() there. I've also added in the dash.